google-cloud-kms/samples/snippets.rb

```diff
@@ -1083,23 +1044,67 @@ def verify_asymmetric_signature_rsa project_id:, location_id:, key_ring_id:, key
     public_key = client.get_public_key name: key_version_name
 
     # Parse the public key.
-    rsa_key = OpenSSL::PKey::RSA.new public_key.pem
+    ec_key = OpenSSL::PKey::EC.new public_key.pem
 
     # Verify the signature.
-    #
-    # Note: The verify_pss() method only exists in Ruby 2.5+.
-    verified = rsa_key.verify_pss "sha256", signature, message, salt_length: :digest, mgf1_hash: "sha256"
+    verified = ec_key.verify "sha256", signature, message
     puts "Verified: #{verified}"
-    # [END kms_verify_asymmetric_signature_rsa]
+    # [END kms_verify_asymmetric_signature_ec]
 
     verified
   end
+
+  if Gem::Version.new(RUBY_VERSION) >= Gem::Version.new("2.5.0")
+    def verify_asymmetric_signature_rsa project_id:, location_id:, key_ring_id:, key_id:, version_id:, message:,
+                                        signature:
+      # [START kms_verify_asymmetric_signature_rsa]
+      # TODO(developer): uncomment these values before running the sample.
+      # project_id  = "my-project"
+      # location_id = "us-east1"
+      # key_ring_id = "my-key-ring"
+      # key_id      = "my-key"
+      # version_id  = "123"
+      # message     = "my message"
+      # signature   = "..."
+
+      # Require the library.
+      require "google/cloud/kms"
+      require "openssl"
+
+      # Create the client.
+      client = Google::Cloud::Kms.key_management_service
+
+      # Build the key version name.
+      key_version_name = client.crypto_key_version_path project:            project_id,
+                                                        location:           location_id,
+                                                        key_ring:           key_ring_id,
+                                                        crypto_key:         key_id,
+                                                        crypto_key_version: version_id
+
+      # Get the public key.
+      public_key = client.get_public_key name: key_version_name
+
+      # Parse the public key.
+      rsa_key = OpenSSL::PKey::RSA.new public_key.pem
+
+      # Verify the signature.
+      #
+      # Note: The verify_pss() method only exists in Ruby 2.5+.
+      verified = rsa_key.verify_pss "sha256", signature, message, salt_length: :digest, mgf1_hash: "sha256"
+      puts "Verified: #{verified}"
+      # [END kms_verify_asymmetric_signature_rsa]
+
+      verified
+    end
+  end
 end
+# rubocop:enable Metrics/ClassLength
 
 if $PROGRAM_NAME == __FILE__
-  methods = (public_methods(false) - [:inspect, :to_s]).sort
+  instance = Snippets.new
+  methods = (instance.public_methods(false) - [:inspect, :to_s]).sort
```

Comment:
I think we can drop the `- [:inspect, :to_s]` since we're now using a custom object instead of the main object so we don't have to exclude those two methods.

Reply:
Done